ios/browser/api/web/ui/chrome_webview.mm

```diff
@@ -130,4 +269,86 @@ - (void)webStateDestroyed:(web::WebState*)webState {
   // never happen.
   NOTREACHED();
 }
+
+// MARK: - CRWWebStateDelegate implementation
+
+- (web::JavaScriptDialogPresenter*)javaScriptDialogPresenterForWebState:
+    (web::WebState*)webState {
+  return dialog_presenter_.get();
+}
+
+// MARK: - ChromeWebControllerUIDelegate implementation
+
+- (void)webController:(ChromeWebViewController*)controller
+    runJavaScriptAlertPanelWithMessage:(NSString*)message
+                               pageURL:(NSURL*)URL
+                     completionHandler:(void (^)(void))handler {
+  UIAlertController* alert =
+      [UIAlertController alertControllerWithTitle:nil
+                                          message:message
+                                   preferredStyle:UIAlertControllerStyleAlert];
+
+  [alert addAction:[UIAlertAction actionWithTitle:@"Ok"
+                                            style:UIAlertActionStyleDefault
+                                          handler:^(UIAlertAction* action) {
+                                            handler();
+                                          }]];
+
+  [controller presentViewController:alert animated:YES completion:nil];
+}
+
+- (void)webController:(ChromeWebViewController*)controller
+    runJavaScriptConfirmPanelWithMessage:(NSString*)message
+                                 pageURL:(NSURL*)URL
+                       completionHandler:(void (^)(BOOL))handler {
+  UIAlertController* alert =
+      [UIAlertController alertControllerWithTitle:nil
+                                          message:message
+                                   preferredStyle:UIAlertControllerStyleAlert];
+
+  [alert addAction:[UIAlertAction actionWithTitle:@"Ok"
```

Comment:
nit: `Ok` should be `OK`, or we should pull a localized string (`IDS_OK` and `IDS_CANCEL` exist in brave's component strings) but not really that necessary